### PR TITLE
[Cosmos] Add AutoScale Header Support 2/2

### PR DESCRIPTION
**Pull request by JericHunter**
Adding the ability to update the autoscale settings

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 1]
```diff
@@ -733,3 +733,19 @@ def _deserialize_throughput(throughput: list) -> Any:
         offer_throughput=throughput,
         properties=throughput_properties[0]
     )
+
+
+def _replace_throughput(throughput: Union[int, ThroughputProperties], new_throughput_properties: list):
+    max_throughput = throughput.auto_scale_max_throughput
+    increment_percent = throughput.auto_scale_increment_percent
+
+    if isinstance(throughput, object):
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:742]: Don't use this - you're already access the attributes on line 739.
This should be using the same kind of logic as your `set_throughput_options` above
Instead try something like:
```python
try:
    if throughput.auto_scale_max_throughput is not None:
        ...
except AttributeError:
    if isinstance(throughput, int)
        ...
    else:
        raise TypeError(...)
```

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 2]
```diff
@@ -733,3 +733,19 @@ def _deserialize_throughput(throughput: list) -> Any:
...
+        if max_throughput is not None:
+            new_throughput_properties['content']['offerAutopilotSettings'][
+                'maxThroughput'] = max_throughput
+            if increment_percent:
+                new_throughput_properties['content']['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy'][
+                    'incrementPercent'] = increment_percent
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:748]: Also need to handle:
```python
            if throughput.offer_throughput:
                new_throughput_properties["content"]["offerThroughput"] = throughput.offer_throughput
```

File: sdk/cosmos/azure-cosmos/test/test_auto_scale.py [hunk 3]
```diff
@@ -59,7 +59,6 @@ def test_auto_scale(self):
         self.assertEqual(
             created_container_properties.auto_scale_max_throughput, 7000)
         self.assertEqual(created_container_properties.auto_scale_increment_percent, 0)
-        self.assertEqual(created_container_properties.offer_throughput, None)
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/test/test_auto_scale.py:62]: Why was this removed?

Comment (JericHunter) [reply sdk/cosmos/azure-cosmos/test/test_auto_scale.py:62]: I messaged you about this previously this was removed because it will not work due to change of logic in deserialize throughput

Comment (annatisch) [reply sdk/cosmos/azure-cosmos/test/test_auto_scale.py:62]: This seems kind of a bug... why does the service return both the autoscale values _and_ a default throughput?

Comment (annatisch) [reply sdk/cosmos/azure-cosmos/test/test_auto_scale.py:62]: As I mentioned in chat - let's perhaps restore the deserialization logic so that we don't have to modify this test in order to unblock this PR/beta release. I think we need to open an issue to explore this more fully before GA.

**Review by annatisch — APPROVED** [review]

**Review by simorenoh — APPROVED** [review]
